api/grpcserver/grpcserver_test.go

```diff
@@ -371,25 +382,29 @@ func (t *ConStateAPIMock) GetNonce(addr types.Address) (types.Nonce, error) {
 	return t.nonces[addr], nil
 }
 
-func (t *ConStateAPIMock) Validation(raw types.RawTx) system.ValidationRequest {
+func (t *ConStateAPIMock) Validation(raw types.RawTx) system.ValidationRequestNew {
 	panic("dont use this")
 }
 
 func NewTx(nonce uint64, recipient types.Address, signer *signing.EdSigner) *types.Transaction {
 	tx := types.Transaction{TxHeader: &types.TxHeader{}}
-	tx.Principal = wallet.Address(signer.PublicKey().Bytes())
+	principal, err := wallet.Address(*signer.PublicKey())
+	if err != nil {
+		panic(err)
+	}
+	tx.Principal = principal
 	if nonce == 0 {
-		tx.RawTx = types.NewRawTx(wallet.SelfSpawn(signer.PrivateKey(),
-			0,
-			sdk.WithGasPrice(0),
-		))
+		tx2, err := wallet.Spawn(signer.PrivateKey(), 0, sdk.WithGasPrice(0))
+		if err != nil {
+			panic(err)
+		}
+		tx.RawTx = types.NewRawTx(tx2)
 	} else {
-		tx.RawTx = types.NewRawTx(
-			wallet.Spend(signer.PrivateKey(), recipient, 1,
-				nonce,
-				sdk.WithGasPrice(0),
-			),
-		)
+		tx2, err := wallet.Spend(signer.PrivateKey(), recipient, 1, nonce, sdk.WithGasPrice(0))
+		if err != nil {
+			panic(err)
+		}
+		tx.RawTx = types.NewRawTx(tx2)
```

Comment:
As `NewTx` is a test-only function, it could take `tb testing.TB` and then this could be simplified to:
```suggestion
		tx2, err := wallet.Spend(signer.PrivateKey(), recipient, 1, nonce, sdk.WithGasPrice(0))
		require.NoError(tb, err)
		tx.RawTx = types.NewRawTx(tx2)
```